// vue2-lay-editor/config/index.js
function EditorConfig() {
  this.themeColor = "#1e80ff";
  this.dialogue = [
    //  {
    //     leftImg:"/dialogue/hamster-1.png",
    //     rightImg:"/dialogue/hamster-2.png"
    //   },
    //   {
    //     leftImg:"/dialogue/hamster-3.png",
    //     rightImg:"/dialogue/hamster-2.png"
    //   },
    //   {
    //     leftImg:"/dialogue/hamster-4.png",
    //     rightImg:"/dialogue/hamster-2.png"
    //   },
    //   {
    //     leftImg:"/dialogue/hamster-5.png",
    //     rightImg:"/dialogue/hamster-2.png"
    //   }
  ];
  // 大图表情包
  this.maxEmoticon = [
    // {
    //     src: 'http://localhost:9080/img/0f85190636580a58617d68e329786dd.d4f2eb16.jpg'
    // },
    // {
    //     src: 'http://localhost:9080/img/0f85190636580a58617d68e329786dd.d4f2eb16.jpg'
    // },
  ];
  // 小图表情包
  this.minEmoticon = [
    // {
    //     title: '微信表情包',
    //     list: [
    //         {
    //             src: "https://baidubox-emoji.cdn.bcebos.com/imgs/%5B%E5%BE%AE%E7%AC%91%5D.png?r=2023060313",
    //         }
    //     ],
    // }
  ];
  this.uploadImg = null;
  // (file, src) => {
  //     return new Promise((resolve, reject) => {
  //         setTimeout(() => {
  //             resolve([undefined, src])
  //         }, 2000)
  //     })
  // }
  this.tools = {};
}

EditorConfig.prototype.setConfig = function (data) {
  this.tools = data.tools || {};
  this.dialogue = data.dialogue || [];
  this.maxEmoticon = data.maxEmoticon || [];
  this.minEmoticon = data.minEmoticon || [];
  this.uploadImg = data.uploadImg
    ? async (file, src) => {
        try {
          let result = await data.uploadImg(file, src);
          return [undefined, result];
        } catch (err) {
          return [err, undefined];
        }
      }
    : null;
};

export default new EditorConfig();
